src/go/k8s/pkg/resources/resource.go

```diff
@@ -0,0 +1,20 @@
+package resources
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Resource interface {
+	Obj() (client.Object, error)
+	Key() types.NamespacedName
+	Kind() string
+}
+
+type OwnedResource interface {
```

Comment:
What is the idea behind the name "Owned"? It seems `Ensure` gets/creates the resource. Could we have a `Resource` that doesn't implement `Ensure`?

Reply:
ha thanks! I was playing with a bit more complicated design at one point, I think for now we're fine with one interface 👍